autolab/Autolab
# Sort YAML output

## Description
Define `sort_hash` which recursively sorts a hash. Then use it when dumping YAML to serialize an assessment's settings.

## Motivation and Context
Currently, the YAML generated after editing an assessment is unordered. This makes `git diff`s noisy, making it harder for instructors who use version control to spot the changes being made.

Note: There are more robust solutions out there, e.g.
- https://www.jvt.me/posts/2020/08/24/sort-recursive-hash-ruby/
- https://www.lucidchart.com/techblog/2016/08/31/deeply-sort-nested-ruby-arrays-and-hashes/

... but the current solution works for our purposes

Closes #1627.

## How Has This Been Tested?
- Note down (i.e. copy and paste elsewhere) an assessment's config file (`courses/<course>/<asmt>/<asmt>.yml`)
- Edit the assessment's settings (e.g. description)
- Observe that the config file is now sorted
- Verify that the contents before and after are the same (modulo the setting changed), except that it is now sorted
- Ensure that reloading config file still works

## Types of changes

## Checklist:
